dpbench/infrastructure/benchmark.py

```diff
@@ -677,23 +686,22 @@ def _set_reference_implementation(self, impl_fnlist):
         Args:
             impl_fnlist : A list of (name, value) pair that represents the name
             of an implementation function and a corresponding function object.
+        Returns:
+            BenchmarkImplFn: The reference benchmark implementation.
+
         """
 
-        ref_impl_fn = None
+        ref_impl = None
 
-        python_impl_fn = [
-            impl_fn for impl_fn in impl_fnlist if "python" in impl_fn[0]
-        ]
-        numpy_impl_fn = [
-            impl_fn for impl_fn in impl_fnlist if "numpy" in impl_fn[0]
-        ]
+        python_impl = [impl for impl in impl_fnlist if "python" in impl.name]
+        numpy_impl = [impl for impl in impl_fnlist if "numpy" in impl.name]
 
-        if numpy_impl_fn:
-            ref_impl_fn = numpy_impl_fn[0]
```

Comment:
The existing code in main does not seem to be correct. The function is setting the reference implementation function (`ref_impl_fn`) to a string that by itself is fine. However, in the only use of `ref_impl_fn` inside the `_get_validation_data` we are doing `ref_impl_fn[0]`.